### PR TITLE
Catty-163: New value overwrites most recent value for Broadcast

### PR DESCRIPTION
**Pull request by Bomml**
*Please enter a short description of your pull request and add a reference to the Jira ticket.*

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

A project variable is created, which stores all recently created messages until the project is closed, i.e. the user goes back to the main screen or the project listing screen. Once the project is reopened, only the used messages are shown.

### CODE REVIEW

File: src/Catty/DataModel/Project/Project.m [hunk 1]
```diff
@@ -433,8 +433,9 @@ + (BOOL)projectExistsWithProjectID:(NSString*)projectID
 
 + (instancetype)defaultProjectWithName:(NSString*)projectName projectID:(NSString*)projectID
 {
-    projectName = [Util uniqueName:projectName existingNames:[[self class] allProjectNames]];
     Project *project = [[Project alloc] init];
+    project.allBroadcastMessages = [[NSMutableOrderedSet alloc] init];
```
Comment (m-herold) [inline src/Catty/DataModel/Project/Project.m:437]: Please initialize the set in the default initializer.

File: src/Catty/DataModel/Project/Project.m [hunk 2]
```diff
@@ -490,6 +491,7 @@ + (nullable instancetype)projectWithLoadingInfo:(ProjectLoadingInfo*)loadingInfo
     }
     project.header.programName = loadingInfo.visibleName;
     project.header.programID = loadingInfo.projectID;
+    project.allBroadcastMessages = [[NSMutableOrderedSet alloc] init];
```
Comment (m-herold) [inline src/Catty/DataModel/Project/Project.m:494]: Please initialize the set in the default initializer.

File: src/Catty/Helpers/Util/Util.m [hunk 3]
```diff
@@ -618,27 +618,26 @@ + (Look*)lookWithName:(NSString*)objectName forObject:(SpriteObject*)object
...
-+ (NSArray*)allMessagesForProject:(Project*)project
++ (NSMutableOrderedSet*)allMessagesForProject:(Project*)project
 {
-    NSMutableArray *messages = [[NSMutableArray alloc] init];
     for(SpriteObject *object in project.objectList) {
         for(Script *script in object.scriptList) {
             if([script isKindOfClass:[BroadcastScript class]]) {
                 BroadcastScript *broadcastScript = (BroadcastScript*)script;
-                [messages addObject:broadcastScript.receivedMessage];
+                [project.allBroadcastMessages addObject:broadcastScript.receivedMessage];
```
Comment (m-herold) [inline src/Catty/Helpers/Util/Util.m:627]: Please do not change global state in a getter function. Suggestion: leave the local `messages` but change to set, and simply add all values of `project.allBroadcastMessages` to this set.

File: src/Catty/ViewController/Continue&New/MaintainObject/MaintainScript/ScriptCollectionViewController.m [hunk 4]
```diff
@@ -1178,7 +1179,7 @@ - (void)updateBrickCellData:(id<BrickCellDataProtocol>)brickCellData withValue:(
                                         minInputLength:kMinNumOfMessageNameCharacters
                                         maxInputLength:kMaxNumOfMessageNameCharacters
                               invalidInputAlertMessage:kLocalizedMessageAlreadyExistsDescription
-                                         existingNames:[Util allMessagesForProject:self.object.project]];
+                                         existingNames:(NSArray*)[Util allMessagesForProject:self.object.project]];
```
Comment (m-herold) [inline src/Catty/ViewController/Continue&New/MaintainObject/MaintainScript/ScriptCollectionViewController.m:1182]: Do not cast the set, but make use of the built-in `array` instance property: https://developer.apple.com/documentation/foundation/nsorderedset/1411531-array

File: src/Catty/Helpers/Util/Util.m [hunk 5]
```diff
@@ -222,7 +222,7 @@ + (void)askUserForUniqueNameAndPerformAction:(SEL)action
                               minInputLength:(NSUInteger)minInputLength
                               maxInputLength:(NSUInteger)maxInputLength
                     invalidInputAlertMessage:(NSString*)invalidInputAlertMessage
-                               existingNames:(NSArray*)existingNames {
+                               existingNames:(NSOrderedSet*)existingNames {
```
Comment (m-herold) [inline src/Catty/Helpers/Util/Util.m:225]: Is there a reason why you changed this parameter? This has not been changed in the interface and will probably lead to some problems, please revert this change.

File: src/CattyTests/Utils/UtilTests.swift [hunk 6]
```diff
@@ -152,4 +162,37 @@ final class UtilTests: XCTestCase {
...
+        broadcastBrick.broadcastMessage = "thirdValue"
+
+        broadcastWaitBrick = BroadcastWaitBrick()
+        script.brickList.add(broadcastWaitBrick!)
+        broadcastWaitBrick.broadcastMessage = "fourthValue"
+
+        let messages = Util.allMessages(for: project)
+
+        XCTAssertEqual(messages?.count, 4)
+    }
```
Comment (m-herold) [inline src/CattyTests/Utils/UtilTests.swift:197]: Please add an additional test case verifying that two same entries added to `allBroadcastMessages` will only return one unique result (without duplicates).

File: src/Catty/Helpers/Util/Util.m (hunk 5 above)
Comment (Bomml) [reply src/Catty/Helpers/Util/Util.m:225]: Oh this was my fault. I don't remember why I changed that. I changed it back.

File: src/CattyTests/Utils/UtilTests.swift (hunk 6 above)
Comment (Bomml) [reply src/CattyTests/Utils/UtilTests.swift:197]: Done :D
